packages/theme/src/cli/services/profile.ts

```diff
@@ -0,0 +1,51 @@
+import {openURL} from '@shopify/cli-kit/node/system'
+import {ensureAuthenticatedStorefront} from '@shopify/cli-kit/node/session'
+import {joinPath, resolvePath, dirname} from '@shopify/cli-kit/node/path'
+import {writeFile} from 'fs/promises'
+import {tmpdir} from 'os'
+import {fileURLToPath} from 'url'
+
+export async function profile(password: string | undefined, storeDomain: string, urlPath: string, asJson: boolean) {
+  // Fetch the profiling from the Store
+  const url = new URL(`https://${storeDomain}/${urlPath}`)
+  url.searchParams.append('profile_liquid', '1')
+  const storefrontToken = await ensureAuthenticatedStorefront([], password)
+  const response = await fetch(url, {
+    headers: {
+      Authorization: `Bearer ${storefrontToken}`,
+    },
+  })
+  const profileJson = await response.text()
+
+  if (asJson) {
+    // Print the JSON
+    process.stdout.write(profileJson)
+  } else {
+    await openProfile(profileJson)
+  }
+}
+
+async function openProfile(profileJson: string) {
+  // Adapted from https://github.com/jlfwong/speedscope/blob/146477a8508a6d2da697cb0ea0a426ba81b3e8dc/bin/cli.js#L63
+  const __filename = fileURLToPath(import.meta.url)
+  const __dirname = dirname(__filename)
+  let urlToOpen = `file://${resolvePath(__dirname, '../../../node_modules/speedscope/dist/release/index.html')}`
```

Comment:
Will this file will be available after building with vite? My understanding is that it would not, since vite will bundle everything and `node_modules` will not exist. May need to import it then use `import.meta.url` or something.